sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py

```diff
@@ -84,12 +126,46 @@ class EventHubProducerClient(ClientBaseAsync):
             :caption: Create a new instance of the EventHubProducerClient.
     """
 
+    @overload
     def __init__(
         self,
         fully_qualified_namespace: str,
         eventhub_name: str,
         credential: "CredentialTypes",
-        **kwargs
+        *,
+        buffered_mode: Literal[False] = False,
+        **kwargs: Any
+    ) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        credential: "CredentialTypes",
+        *,
+        buffered_mode: Literal[True],
+        on_error: Callable[[SendEventTypes, Optional[str], Exception], Awaitable[None]],
+        on_success: Callable[[SendEventTypes, Optional[str]], Awaitable[None]],
+        max_buffer_length: int = 1500,
+        max_wait_time: float = 1,
+        **kwargs: Any
+    ) -> None:
+        ...
+
+    def __init__(
+        self,
+        fully_qualified_namespace: str,
+        eventhub_name: str,
+        credential: "CredentialTypes",
+        *,
+        buffered_mode: bool = False,
+        on_error: Optional[Callable[[SendEventTypes, Optional[str], Exception], Awaitable[None]]] = None,
+        on_success: Optional[Callable[[SendEventTypes, Optional[str]], Awaitable[None]]] = None,
+        max_buffer_length: Optional[int] = None,
+        max_wait_time: Optional[float] = None,
+        **kwargs: Any
```

Comment:
wondering why typehints specified for the constructor implementation here but not in sync?

Reply:
updated, I forgot to update it as previously we are using comment style typing :P